### PR TITLE
Some minor tweaks/fixes

### PR DESCRIPTION
**Pull request by crowder**
I found that sometimes tinkering with memory is fragile and my creeps lose their roles. Since you thoughtfully put role-names into the creep's name, they can be re-calculated there, so I added that.

Added a couple spaces in spots, and an output to show a creep's body at the time they spawn also (I find this useful as I dynamically calculate bodies at spawn time).

I moved the grunt-screeps credentials data out of Gruntfile.js and into a separate local (ignored) file so that I don't accidentally publish my screeps credentials when I'm issuing a PR.

I found that creating an extensions/krelin directory as a git repo was prone to problems in codeGen, so I fixed the codegen to robustly (I think?) detect valid identifiers before marking those directories for inclusion. It will quietly complain about those directories as they're being skipped, as well.

### CODE REVIEW

**Comment by crowder** [comment]
This is failing Travis CI because of the missing .screeps-creds file...  not sure how best to handle that.

File: .gitignore [hunk 1]
```diff
@@ -30,3 +30,5 @@ node_modules
 
 # Users Environment Variables
 .lock-wscript
+
+.screeps-creds
```
Comment (avdg) [inline .gitignore]: Hmm I still need a good solution for this. Even though I have to fix it, I don't like to see these things in my commits...

**Comment by avdg** [comment]
Yeah, I was doubting myself about getting a default file or something.

Also, it might be easy to commit small changes by doing `git add -p [optional files (without it checks all)]`.

**Comment by avdg** [comment]
About the configuration options: I will add an option to merge a config like using a file check and the api described at http://gruntjs.com/api/grunt.config#grunt.config.merge. I'm not going to merge the config setup as done in this pr though.

About the role fixes: Many fixes are very obvious. Will check later to merge those fixes in.

**Comment by avdg** [comment]
About extensions: try following

``` js
{
    "screepsCodeGenerator": {
        "options": {
            "extensions": [
                path.join(__dirname, 'extensions'),
                path.join(__dirname, '../screeps-other-repo/extensions')
            ],
        }
    }
}
```

That way you can separate your custom extensions from the main repo. You can also maintain the git repo there without any problems.

**Comment by avdg** [comment]
Private settings are in f769b31348837fa4f8a37430dcce35ae9ae858d9

**Comment by avdg** [comment]
Cherry picked ff4b5ba658071 for now, see 274736fbf5b173928303f

**Comment by avdg** [comment]
Also, I have to update the readme soon to update the settings set up section.

**Comment by crowder** [comment]
No interest in 9c53a28?

Also, I still recommend keeping 0576844 or some variation like it, (or perhaps bail with an error message), since it can be difficult to track down the cause of the issue. You end up with an object literal containing a property with an invalid identifier name. In my case, that invalid name was ".git", but it could be anything that's not a legitimate identifier.

**Comment by avdg** [comment]
- 9c53a28: Very likely to rewrite it soon a little bit (using indexof and substr)
- 0576844: Obviously there is a bug in the code generator here, but also there is already a policy for file names starting with a dot, and that is to simply ignore them. I might extend this policy to directories as well. See https://github.com/avdg/screeps/blob/6e89da9f1bc84afb24a8ba09ea35a1ba15541454/README.md#extensions-folder-structure

Later (but don't ask when) I plan to add _ prefix support to mark the content to be treated as queue elements. That way, certainly for library extensions, I can avoid naming conflicts when there is a big growth of packages. That way, there is less need to make packages compatible on things where conflicts don't matter anyway. Roles, procedures, commands will stay as they are, as conflicts are bad anyway (the key name is used to refer to them anyway).

I will probably work again on this stuff very soon, and will try to get as much as I can out of this ticket before moving on to other stuff.

File: extensions/tools/scripts/creeps.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ module.exports = function() {
 
         if (typeof creep.memory.role !== "string") {
             console.log('Warning: Creep without role at ' + creep.pos.x + "," + creep.pos.y);
+            creep.memory.role = creep.name.split(' ')[0];
```
Comment (avdg) [inline extensions/tools/scripts/creeps.js:13]: Maybe I'll let a hook handle this, so the behavior can be modified or overwritten.

**Comment by avdg** [comment]
bad625a3af4ca2e8089c

Files starting with a '.' are now skipped. Though I fail to understand why some lines are not covered, since removing them results in failing tests...

Talking about these lines
- https://coveralls.io/builds/3336991/source?filename=lib%2Fcodegen%2Fextensions.js#L251
- https://coveralls.io/builds/3336991/source?filename=lib%2Fcodegen%2Fextensions.js#L291

**Comment by avdg** [comment]
Cause of not covered files: ignored files are not copied to lib-cov. Digging in the cause...

**Comment by avdg** [comment]
Current todo list:
- [x] Fix codegen handling variables that don't fit properly without quotes (in 6725dc7b60baf10dc4d4a4338a21a47c90590c2f, a137ce3c1df9ca7633273b03d4db2b0213998494 and e1d0ee3730bd57a66631ab1867a4f1e40dadf6aa)
- [x] Creeps without role recovery (in 1df4ddb52f10870daac5a0865fd92ff8e8a423e4 and aca79fecdfaca5c1e4fed34746a940e90e32fe51)
- [x] Fix copy files to lib-cov so it includes all the files inside the mock directory (in 3befc35b7ca806934d7c1d4bed)
- [x] Update readme (in e1f4c04f0bf527bad8ee50ded1ac445336cc76a3, 623c3718e1e6d675f6ead2fc49739032110b98cb and c55caa828ee3f8ea58f9800dcd5900b95a415e99)

**Comment by avdg** [comment]
All tweaks from this pull request should be in the repository in some form by now, closing ticket.

**Comment by avdg** [comment]
Ow yeah, 9c53a28958cf5c41ed06f4e70054669b8ac83b0c isn't completely in the repo. Not sure if I should add this as default behavior or not...

**Pull request closed without merging**